### PR TITLE
Abstract the strnlen function for environments that do not have it (e.g., Solaris 10)

### PR DESCRIPTION
**Pull request by rhc54**
Fixes https://github.com/open-mpi/ompi/issues/1771

### CODE REVIEW

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by jsquyres** [comment]
@siegmargross This should fix the strnlen problem.

**Comment by ibm-ompi** [comment]
Build Failed with GNU compiler! Please review the log, and get in touch if you have questions.

**Comment by ibm-ompi** [comment]
Build Failed with XL compiler! Please review the log, and get in touch if you have questions.

File: opal/mca/pmix/pmix114/pmix/src/util/strnlen.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * Buffer strnlen function for portability to archaic platforms.
+ */
+
+#ifndef PMIX_STRNLEN_H
+#define PMIX_STRNLEN_H
+
+#include <src/include/pmix_config.h>
+
+#if defined(HAVE_STRLEN)
```
Comment (jsquyres) [inline opal/mca/pmix/pmix114/pmix/src/util/strnlen.h]: @rhc54 Should this be HAVE_STRNLEN (i.e., missing the "N")?

File: opal/mca/pmix/pmix114/pmix/src/util/strnlen.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+#if defined(HAVE_STRNLEN)
+#define PMIX_STRNLEN(c, a, b)       \
+    (c) = (int)strnlen(a, b)
+#else
+#define PMIX_STRNLEN(c, a, b)           \
+    do {                                \
+        int _x;                         \
+        (c) = 0;                        \
+        for (_x=0; _x < (b); _x++) {    \
+            if ('0' == (a)[_x]) {       \
```
Comment (jsquyres) [inline opal/mca/pmix/pmix114/pmix/src/util/strnlen.h]: Sorry -- another typo -- shouldn't this be `'\0'` (note the missing slash)?

Comment (rhc54) [inline opal/mca/pmix/pmix114/pmix/src/util/strnlen.h]: maybe it would have been faster if you had just written it 😃

**Comment by rhc54** [comment]
bot:retest

**Comment by rhc54** [comment]
Looks like travis was unable to setup one of its environments - not associated with these changes

**Pull request merged**